### ISSUE TITLE
Infinite `PATH` manipulation/growth

### ISSUE THREAD

**Issue by 13k** [onivim/oni#1706]
I was trying to debug why the go language server (`go-langserver`) wasn't being started, when I saw a peculiar behavior regarding `PATH` manipulation.

When I first started Oni, `process.env.PATH` contained 44 entries:

```
> process.env.PATH.split(':').length
44
```

After opening a bunch of `.go` files, it grew to 176:

```
> process.env.PATH.split(':').length
176
```

If I continue to open files, the count keeps growing indefinitely.
It's worth noting that the the lang server failed to start (maybe it only occurs in this situation):

```
[LanguageClientProcess]: Starting process via 'go-langserver'
bundle.js:1 Error: spawn go-langserver ENOENT
    at exports._errnoException (util.js:1024)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:192)
    at onErrorNT (internal/child_process.js:374)
    at _combinedTickCallback (internal/process/next_tick.js:138)
    at process._tickCallback (internal/process/next_tick.js:180)
t.error @ bundle.js:1
a.on @ 0.bundle.js:64
emitOne @ events.js:115
emit @ events.js:210
ChildProcess._handle.onexit @ internal/child_process.js:198
onErrorNT @ internal/child_process.js:374
_combinedTickCallback @ internal/process/next_tick.js:138
_tickCallback @ internal/process/next_tick.js:180
0.bundle.js:182 Uncaught (in promise) Error: Unable to start language server process
    at h.<anonymous> (0.bundle.js:182)
    at Generator.next (<anonymous>)
    at a (0.bundle.js:182)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188)
```

**Comment by 13k** [comment]
> Which platform is this? OSX or Linux?

Linux

> Is this with the latest binary release

Latest binary, sorry, I didn't test latest master. I can do it and get back to you if you need

**Comment by 13k** [comment]
In regards to #1665, it can be related indeed.

My 2 cents: `process.env` should be immutable and customizations should be done via options passed downstream.

**Comment by bryphe** [comment]
> My 2 cents: process.env should be immutable and customizations should be done via options passed downstream.

Definitely! 💯 

FYI, I was able to reproduce on the released version - 0.3.0 - and not on `master`. Prior to #1665 , everytime we'd spin up a process, it'd mutate `process.env` - adding additional items to the path.

The next step would be to figure out why `go-langserver` isn't starting up for you - have you tried explicitly setting `language.go.languageServer.command` in your config.js?

For example, I have:
```
"language.go.languageServer.command": "/Users/bryphe/go/bin/go-langserver",
```

in my config, since it isn't in my `path`.

**Comment by 13k** [comment]
> figure out why go-langserver isn't starting up for you

I did, my env was not set up properly. I'm using `gimme` to switch go versions and starting `oni` from the command line, but forgot to switch it for that specific terminal. After switching, everything worked.  :man_facepalming:

> I was able to reproduce on the released version - 0.3.0 - and not on master.

It's fixed then, good to know. I think this one can be closed :)

Thanks for your time!

**Comment by Akin909** [comment]
Gonna close this @13k and @bryphe since its resolved 👍
